### PR TITLE
CLOUD-1740 add linkerd inject interceptor

### PR DESCRIPTION
**Pull request by svenwltr**
> https://jira.rebuy.de/browse/CLOUD-1740

### CODE REVIEW

**Comment by svenwltr** [comment]
@rebuy-de/prp-kubernetes-deployment Please review.

File: Dockerfile [hunk 1]
```diff
@@ -12,11 +12,17 @@ RUN mkdir -p ${GOPATH}/src ${GOPATH}/bin
 # Install Go Tools
 RUN GO111MODULE= go get -u golang.org/x/lint/golint
 
+# Install Linkerd
+RUN curl -Lo /usr/local/bin/linkerd https://github.com/linkerd/linkerd2/releases/download/stable-2.1.0/linkerd2-cli-stable-2.1.0-linux
```
Comment (bjoernhaeuser) [inline Dockerfile]: can only recommend to verify the checksum

Comment (svenwltr) [reply Dockerfile]: I thought about this, but what would us protect this from?

Comment (bjoernhaeuser) [reply Dockerfile]: broken downloads :)

File: pkg/api/app.go [hunk 2]
```diff
@@ -9,6 +9,7 @@ import (
 	"github.com/rebuy-de/kubernetes-deployment/pkg/gh"
 	"github.com/rebuy-de/kubernetes-deployment/pkg/interceptors"
 	"github.com/rebuy-de/kubernetes-deployment/pkg/interceptors/annotater"
+	"github.com/rebuy-de/kubernetes-deployment/pkg/interceptors/injecter"
```
Comment (der-eismann) [inline pkg/api/app.go]: injector. :nothingtodohere:

File: Dockerfile (hunk 1 above)
Comment (svenwltr) [reply Dockerfile]: fixed

File: pkg/api/app.go (hunk 2 above)
Comment (svenwltr) [reply pkg/api/app.go]: fixed

File: pkg/api/app.go [hunk 3]
```diff
@@ -113,6 +114,13 @@ func (app *App) StartInterceptors(service *settings.Service) {
 		}).Debug("enabling annotater interceptor")
 		app.Interceptors.Add(annotater.New())
 	}
+
+	if interceptors.Injecter.Enabled == settings.Enabled {
```
Comment (der-eismann) [inline pkg/api/app.go]: Injector!!

File: pkg/settings/interceptors.go [hunk 4]
```diff
@@ -11,6 +11,7 @@ type Interceptors struct {
 	RemoveOldJob        Interceptor                `yaml:"removeOldJob"`
 	Waiter              Interceptor                `yaml:"waiter"`
 	Annotater           Interceptor                `yaml:"annotater"`
+	Injecter            Interceptor                `yaml:"injector"`
```
Comment (der-eismann) [inline pkg/settings/interceptors.go]: Injector!!

File: pkg/api/app.go (hunk 3 above)
Comment (svenwltr) [reply pkg/api/app.go]: dammit! fixed!

File: pkg/settings/interceptors.go (hunk 4 above)
Comment (svenwltr) [reply pkg/settings/interceptors.go]: dammit! fixed!

**Comment by svenwltr** [comment]
bump

**Pull request merged**